src/qml/editorwidgets/ValueRelation.qml

```diff
@@ -63,111 +63,143 @@ EditorWidgetBase {
-          anchors {
-            horizontalCenter: parent.horizontalCenter
-            verticalCenter: parent.verticalCenter
-          }
-          font.bold: true
-          font.pointSize: Theme.resultFont.pointSize
-          color: Theme.mainTextColor
-          text: section
-          visible: listModel.displayGroupName
-        }
-      }
-
-      delegate: Item {
-        id: listItem
-        anchors {
-          left: parent ? parent.left : undefined
-          right: parent ? parent.right : undefined
-        }
-        height: Math.max(valueRelationList.itemHeight, valueText.height)
-
+      GridLayout {
+        id: valueGridView
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.top: parent.top
         focus: true
+        columns: config['NofColumns'] ? config['NofColumns'] : 1
+        columnSpacing: 1
+        rowSpacing: 0
+
+        Repeater {
+          id: repeater
+          model: listModel
+
+          delegate: Item {
+            id: listItem
+            Layout.fillWidth: true
+            Layout.fillHeight: true
+            Layout.minimumHeight: Math.max(valueText.height, valueRelationList.itemHeight) + (header.visible ? header.height : 0)
+
+            property string groupFieldVal: groupFieldValue ? groupFieldValue : ""
+            property bool rowContainsHeader: false
+            property alias headerItem: header
+
+            focus: true
+
+            Component.onCompleted: {
+              // Check if any item in the current row has a visible header item.
+              // If found, mark the current listItem's row as containing a header.
+              const row = Math.floor(index / valueGridView.columns);
+              const start = row * valueGridView.columns;
+              const end = Math.min(start + valueGridView.columns, repeater.count - 1);
+              for (let i = start; i < end; ++i) {
+                if (repeater.itemAt(i) && repeater.itemAt(i).headerItem.visible) {
+                  listItem.rowContainsHeader = true;
+                  break;
+                }
+              }
+            }
 
-        Rectangle {
-          id: checkBoxRow
-          width: parent.width
-          height: listItem.height
-          color: Theme.mainBackgroundColor
-
-          CheckDelegate {
-            id: checkBox
-            anchors.verticalCenter: parent.verticalCenter
-            anchors.left: parent.left
-            width: valueRelationList.itemHeight
-            height: valueRelationList.itemHeight
-            enabled: isEnabled
-
-            checked: model.checked
-
-            indicator.height: 16
-            indicator.width: 16
-            indicator.implicitHeight: 24
-            indicator.implicitWidth: 24
-          }
+            Rectangle {
+              id: header
+              visible: listModel.groupField && groupFieldValue !== "" && (index === 0 || (index > 0 ? groupFieldValue !== repeater.itemAt(index - 1).groupFieldVal : false))
```

Comment:
I'd transform that into a property bool isVisible: blah blah blah , then visible: isVisible -- then use isVisible elsewhere (eg with the height property).

Visible is a tricky property.